### PR TITLE
Auth samples - added HTTP samples

### PR DESCRIPTION
**Pull request by ryanmats**

### CODE REVIEW

File: auth/src/auth_http_explicit.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    # Create an HTTP Client using Guzzle and pass in the credentials.
+    $http_client = new Client([
+        'handler' => $stack,
+        'base_uri' => 'https://www.googleapis.com/storage/v1/',
+        'auth' => 'google_auth'
+    ]);
+
+    # Make an authenticated API request (listing storage buckets)
+    $response = $http_client->request('GET', 'b', ['query' =>['project'=>$projectId]]);
```
Comment (bshaffer) [inline auth/src/auth_http_explicit.php]: Let's format this a bit better,

```php
$query = ['project' => $projectId];
$response = $http_client->request('GET', 'b', [
    'query' => $query
]);
```

File: auth/src/auth_http_implicit.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    # Create a HTTP Client using Guzzle and pass in the credentials.
+    $http_client = new Client([
+        'handler' => $stack,
+        'base_uri' => 'https://www.googleapis.com/storage/v1/',
+        'auth' => 'google_auth'
+    ]);
+
+    # Make an authenticated API request (listing storage buckets)
+    $response = $http_client->request('GET', 'b', ['query' =>['project'=>$projectId]]);
```
Comment (bshaffer) [inline auth/src/auth_http_implicit.php]: same!

File: auth/src/auth_http_explicit.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    $body_content = json_decode($response->getBody());
```
Comment (bshaffer) [inline auth/src/auth_http_explicit.php]: nit, but let's add a cast to string here. That way the conversion is explicit:

```php
$body_content = json_decode((string) $response->getBody());
```

File: auth/src/auth_http_implicit.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    $body_content = json_decode($response->getBody());
```
Comment (bshaffer) [inline auth/src/auth_http_implicit.php]: same!

File: auth/src/auth_http_explicit.php (hunk 1 above)
Comment (ryanmats) [inline auth/src/auth_http_explicit.php]: done

File: auth/src/auth_http_explicit.php (hunk 3 above)
Comment (ryanmats) [inline auth/src/auth_http_explicit.php]: done

File: auth/src/auth_http_implicit.php (hunk 2 above)
Comment (ryanmats) [inline auth/src/auth_http_implicit.php]: done

File: auth/src/auth_http_implicit.php (hunk 4 above)
Comment (ryanmats) [inline auth/src/auth_http_implicit.php]: done

**Pull request merged**